components/squad/squad_row.lua

```diff
@@ -102,16 +111,21 @@ function SquadRow:newteam(args)
 		if mw.ext.TeamTemplate.teamexists(newteam) then
 			cell:wikitext(mw.ext.TeamTemplate.team(args.newteam:lower(),
 				args.newteamdate or ReferenceCleaner.clean(args.leavedate)))
+		elseif self.options.useTemplatesForSpecialTeams then
+			if Table.includes(SquadRow.specialTeamsTemplateMapping, newteam) then
+				cell:wikitext(Template.safeExpand(mw.getCurrentFrame(), SquadRow.specialTeamsTemplateMapping[newteam]))
```

Comment:
this doesn't work
`Table.includes` checks against the values not the keys
```suggestion
		elseif self.options.useTemplatesForSpecialTeams then
			local newTeamTemplate = SquadRow.specialTeamsTemplateMapping[newteam]
			if newTeamTemplate then
				cell:wikitext(Template.safeExpand(mw.getCurrentFrame(), newTeamTemplate))
```
also kick line 6 with this change

(tested it with your modules on RL)

Reply:
Cool, done